expression/expression.go

```diff
@@ -269,6 +270,22 @@ func TableInfo2Schema(tbl *model.TableInfo) Schema {
 		}
 		schema.Append(newCol)
 	}
+	for _, idx := range tbl.Indices {
```

Comment:
For primary key with integer type, it will not appear in tbl.Indices. You should check column.PkAsHandle attribute.